crates/llm-base/src/lib.rs

```diff
@@ -64,16 +64,24 @@ pub struct InferenceParameters {
     /// The number of tokens to consider for the repetition penalty.
     pub repetition_penalty_last_n: usize,
 }
-impl Default for InferenceParameters {
-    fn default() -> Self {
+impl InferenceParameters {
+    /// Returns a reasonable default for the parameters.
+    ///
+    /// Note that these parameters are not necessarily optimal for all models, and that
+    /// you may want to tweak them for your use case.
+    ///
+    /// This is intentionally not a `Default` implementation. The values specified here may change
+    /// in the future, and we want to make sure that users are aware of this and do not accidentally
+    /// rely on the values.
+    pub const fn reasonable_default() -> Self {
```

Comment:
I see where you're coming from and I'm not super-sure about this myself, but I wanted to deliberately disable the use of `Default` so that the user can't just smash `Default::default` in without thinking about it at all.

But perhaps you're right, and there's not really that much distinction in practice. I'll think about it...

Reply:
IMO the "correct" way to approach it would be to add documentation to `InferenceParameter`, `InferenceRequest` and `infer` that specifies that it's _highly_ discouraged to use default `InferenceParameters` in production/performance-sensitive situations 🤷🏻